spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -621,6 +640,32 @@ public static void importSparkPartitions(
     }
   }
 
+  public static List<SparkPartition> filterPartitions(List<SparkPartition> partitions,
+                                                      Map<String, String> partitionFilter) {
+    if (partitionFilter.isEmpty()) {
+      // No partition filter arg
+
+      if (partitions.isEmpty()) {
+        throw new IllegalArgumentException("Cannot add files, no files found in the table.");
```

Comment:
I feel like this should be checked outside of `filterPartitions`. I think this utility method should focus only on filtering passed partitions. Also, I think the error message can be a bit more precise. In this case, we did not find partitions for the partitioned table rather than did not find files.